### PR TITLE
fileedservice-test.txt file for Issue#13398

### PR DESCRIPTION
**Pull request by b21tilbe**
Created 3 different tests for the file fileedservice.php

### CODE REVIEW

**Review by d21erisu — CHANGES_REQUESTED** [review]
Expected output is missing, i can't see any json data. The part about line 17 can be placed first so that you have everything in order. Otherwise i think it looks good (mind that its afternoon and i'm tired).

**Comment by b21ricni** [comment]
Fileedservice Json data can be found https://cms.webug.se/root/G2/students/(Your student username)/LenaSYS/DuggaSys/fileed.php?courseid=1885&coursevers=1337 if you have set up your server. I can help with this if you come by uni while i'm there since it can be a pain to fix if you are unlucky

File: backend-models/service-tests/fileedservice-tests.txt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+*********
+
+If successful: no output should be displayed.
+If failed: An error message should pop up.
+
+--------------------------------------------------------------------------------------------------------------------------------------------
+Test logging service events (line 17)
+--------------------------------------------------------------------------------------------------------------------------------------------
+
+*********
```
Comment (b21ricni) [inline backend-models/service-tests/fileedservice-tests.txt:98]: *****************
 Sent input
*****************

File: backend-models/service-tests/fileedservice-tests.txt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+if ($result === true) {
+    echo "File deleted successfully.";
+} else {
+    echo "Error deleting file: " . $result;
+}
+
+*********
+* Expected output:
```
Comment (b21ricni) [inline backend-models/service-tests/fileedservice-tests.txt:46]: *****************
 Service output
*****************

** Gather service output *
Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
this is the expected output for the micro service.

Comment (b21ricni) [reply backend-models/service-tests/fileedservice-tests.txt:46]: change to this

File: backend-models/service-tests/fileedservice-tests.txt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+*********
+* Check if the log message was written correctly:
+*********
+
+$expected_log = "[$log_uuid] [ServiceServerStart] [fileedservice.php] [1] DELFILE 123 2023-00-00 1 test.txt 2";
+$logged_message = file_get_contents("path/to/log/test.txt");
+assert($expected_log == $logged_message, "Failed.");
+
+*********
+* Expected output:
```
Comment (b21ricni) [inline backend-models/service-tests/fileedservice-tests.txt:123]: *****************
 Service output
*****************

** Gather service output *
Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
this is the expected output for the micro service.

Comment (b21ricni) [reply backend-models/service-tests/fileedservice-tests.txt:123]: change to this

File: backend-models/service-tests/fileedservice-tests.txt [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        }
+
+*********
+* Check that the actual path is equal to the expected path:
+*********
+
+$this->assertEquals($expectedPath, $actualPath);
+
+*********
+* Expected output:
```
Comment (b21ricni) [inline backend-models/service-tests/fileedservice-tests.txt:88]: Change to:
*****************
 Service output
*****************

** Gather service output *
Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
this is the expected output for the micro service.

**Review by b21ricni — CHANGES_REQUESTED** [review]
And add json data for the functions

**Comment by c21alest** [comment]
Related to: #13398

But this is v4 @b21tilbe you need to copy test description to a new issue branch v5.

**Review by d21erisu — APPROVED** [review]

**Pull request merged**